src/docfx/config/AppData.cs

```diff
@@ -73,5 +77,17 @@ private static string GetAppDataRoot()
                 ? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".docfx")
                 : Path.GetFullPath(docfxAppData);
         }
+
+        private static string GetRepositoryUrlFromEnvironment()
+        {
+            var value = Environment.GetEnvironmentVariable("DOCFX_REPOSITORY_URL");
+            if (!string.IsNullOrEmpty(value))
+            {
+                var (remote, branch) = HrefUtility.SplitGitHref(value);
```

Comment:
don't split href here, the url should be the original url as is. `branch` is coming from another environment variable.